nim_status/multiaccount.nim

```diff
@@ -49,27 +59,45 @@ proc buildAccount(privateKey: PrivateKey): Account =
 
   return acc
 
-proc generateAndDeriveAddresses*(mnemonicPhraseLength: int, n: int,
-  bip39Passphrase: string, paths: seq[string]): seq[MultiAccount] =
+proc deriveAccounts*(multiAcc: MultiAccount, paths: seq[KeyPath]): seq[Account] =
+  var accounts: seq[Account]
+  for p in paths:
+    let skResult = derive(multiAcc.keySeed, p)
+    var acc = buildAccount(cast[PrivateKey](skResult.get()))
```

Comment:
```suggestion
    var acc = buildAccount(PrivateKey(skResult.get()))
```

`cast` in general is an unsafe construct that should be avoided unless you specifically know what it does and why (in which case we tend to comment why it happens to work) - when converting between distinct types, a conversion is used, like suggested

Reply:
All use of `cast` has been removed from modules in `nim_status`. There is still some usage in the tests, will clean that up over time.